WordPress/gutenberg
# Introduce ExperimentalBlockEditorProvider

## Description

This PR introduces `ExperimentalBlockEditorProvider` that prevents mixing experimental editor settings and the public `BlockEditorProvider` component. The actual filtering is handled in a private `__experimentalUpdateSettings` selector in the block-editor store.

Experimental settings may still be set via a PHP plugin – limiting this vector would require a separate PR.

## Rationale

WordPress extenders cannot update the experimental block settings on their own. The `updateSettings()` actions of the `@wordpress/block-editor` store will filter out all the settings that are **not** a part of the public API. The only way to actually store them is via private action. `__experimentalUpdateSettings()`.

To privatize a block editor setting, add it to the `privateSettings` list in 
[/packages/block-editor/src/store/actions.js](/packages/block-editor/src/store/actions.js):

```js
const privateSettings = [
	'__unstableInserterMediaCategories',
	// List a block editor setting here to make it private
];
```

## Mobile apps

This PR updates a few `.native.js` files. Mobile apps choose to import the `.native.js` files over their `.js` counterparts whenever possible. At the same time, this PR introduced the expectation of having two extra named exports: `ExperimentalEditorProvider` and `ExperimentalBlockEditorProvider`. I updated the native files to export the regular Provider component under the experimental name. This works because the settings filtering is restricted to the web – it doesn't make sense in context of mobile apps anyway.

## Testing instructions

* This PR comes with unit tests – confirm they pass and no other CI checks are red
* Confirm the code changes make sense

cc @ntsekouras @youknowriad
